nodestream_plugin_neptune/neptune_connection.py

```diff
@@ -0,0 +1,100 @@
+import json
+from abc import ABC, abstractmethod
+from logging import getLogger
+
+from aiobotocore.session import get_session
+
+
+class NeptuneConnection(ABC):
+    @abstractmethod
+    async def execute(self, query_stmt: str, parameters: list):
+        pass
+
+
+class NeptuneDBConnection(NeptuneConnection):
+    @classmethod
+    def from_configuration(cls, host: str, region: str):
+        return cls(host=host, region=region)
+
+    def __init__(self, host: str, region: str) -> None:
+        self.host = host
+        self.region = region
+        self.session = get_session()
+        self.logger = getLogger(self.__class__.__name__)
+
+    async def execute(self, query_stmt: str, parameters: list):
+        response = None
+        async with self.session.create_client(
+            "neptunedata", region_name=self.region, endpoint_url=self.host
+        ) as client:
```

Comment:
I'm observing that this method and `execute` inside of `NeptuneAnalyticsConnection` as well as here in `NeptuneDBConnection`. We'd need to be very intentional to keep the error handling the same between the two and logging. 

What if we relied on a [template method](https://refactoring.guru/design-patterns/template-method), to delegate opening a `client` and getting a `response` to keep the core behavior the same?